spring-cloud-dataflow-server-cloudfoundry-autoconfig/src/main/java/org/springframework/cloud/dataflow/server/cloudfoundry/config/security/CloudFoundryOAuthSecurityConfiguration.java

```diff
@@ -0,0 +1,106 @@
+import org.springframework.cloud.dataflow.server.cloudfoundry.config.security.support.CloudFoundryDataflowAuthoritiesExtractor;
+import org.springframework.cloud.dataflow.server.cloudfoundry.config.security.support.CloudFoundrySecurityService;
+import org.springframework.cloud.dataflow.server.config.security.support.OnSecurityEnabledAndOAuth2Enabled;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationContextAware;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Conditional;
+import org.springframework.context.annotation.Configuration;
+
+/**
+ * When running inside Cloud Foundry, this {@link Configuration} class will reconfigure Spring Cloud Data Flow's security
+ * setup in {@link OAuthSecurityConfiguration}, so that only users with the
+ * {@link CloudFoundryOAuthSecurityConfiguration#CF_SPACE_DEVELOPER_ROLE} can access the REST APIs.
+ *
+ * Therefore, this configuration will ensure that only Cloud Foundry {@code Space Developers} have access to the
+ * underlying REST API's.
+ *
+ * For this to happen, a REST call will be made to the Cloud Foundry Permissions API via
+ * CloudFoundrySecurityService inside the {@link DefaultDataflowAuthoritiesExtractor}.
+ *
+ * If the user has the respective permissions, the
+ * {@link CloudFoundryOAuthSecurityConfiguration#CF_SPACE_DEVELOPER_ROLE} will be assigned to the user.
+ *
+ * See also: https://apidocs.cloudfoundry.org/258/apps/retrieving_permissions_on_a_app.html
+ *
+ * @author Gunnar Hillert
+ */
+@Configuration
+@Conditional(OnSecurityEnabledAndOAuth2Enabled.class)
+@ConditionalOnCloudPlatform(CloudPlatform.CLOUD_FOUNDRY)
+public class CloudFoundryOAuthSecurityConfiguration {
+
+	private static final Logger logger = LoggerFactory.getLogger(CloudFoundryOAuthSecurityConfiguration.class);
+
+	@Bean
+	@ConditionalOnProperty(name="spring.cloud.dataflow.security.cf-use-uaa", havingValue="true")
+	public CloudFoundryDataflowAuthoritiesExtractor authoritiesExtractor() {
+		return new CloudFoundryDataflowAuthoritiesExtractor(cloudFoundrySecurityService());
+	}
+
+	@Bean
+	@ConditionalOnProperty(name="spring.cloud.dataflow.security.cf-use-uaa", havingValue="true")
+	public UserInfoTokenServicesPostProcessor UserInfoTokenServicesPostProcessor() {
+		UserInfoTokenServicesPostProcessor userInfoTokenServicesPostProcessor = new UserInfoTokenServicesPostProcessor();
+		return userInfoTokenServicesPostProcessor;
+	}
+
+	@Bean
+	@ConditionalOnProperty(name="spring.cloud.dataflow.security.cf-use-uaa", havingValue="true")
+	public CloudFoundrySecurityService cloudFoundrySecurityService() {
+		return new CloudFoundrySecurityService();
+	}
+
+	public class UserInfoTokenServicesPostProcessor implements BeanPostProcessor, ApplicationContextAware {
+
+		private ApplicationContext ctx;
+
+		@Override
+		public Object postProcessBeforeInitialization(Object bean, String beanName) {
```

Comment:
This way of doing feels wrong to me. Should allow overradibility by way of `@ConditionalOnMissingXXX` or something